tests/src/JIT/HardwareIntrinsics/X86/Avx/Add.Double.cs

```diff
@@ -28,32 +28,41 @@ private static void AddDouble()
                 // Validates basic functionality works, using Unsafe.Read
                 test.RunBasicScenario_UnsafeRead();
 
-                // Validates basic functionality works, using Load
-                test.RunBasicScenario_Load();
+                if (Avx.IsSupported)
+                {
+                    // Validates basic functionality works, using Load
+                    test.RunBasicScenario_Load();
 
-                // Validates basic functionality works, using LoadAligned
-                test.RunBasicScenario_LoadAligned();
+                    // Validates basic functionality works, using LoadAligned
+                    test.RunBasicScenario_LoadAligned();
+                }
 
                 // Validates calling via reflection works, using Unsafe.Read
                 test.RunReflectionScenario_UnsafeRead();
 
-                // Validates calling via reflection works, using Load
-                test.RunReflectionScenario_Load();
+                if (Avx.IsSupported)
+                {
```

Comment:
Another question:  How do you feel about checking in the templated code if `test.IsSupported` is the same thing as `{1}.IsSupported` and omit this 2nd check?  It feels odd that we check the same condition twice - once above, and once here - in the cases where they are the same.

Reply:
These "templates" are really just files that go through `string.Format`. It would require quite a bit of rework to support conditional formatting. For context, I had originally tried to use T4, but that did not easily support generating multiple files from a single template (which is our primary scenario). `string.Format` on the other hand made this trivial with the trade-off that the generated code may not be "optimal" (which is also fine, since it is just test code).